pycbc/waveform/compress.py

```diff
@@ -726,17 +749,28 @@ def write_to_hdf(self, fp, template_hash, root=None):
             Put the `compressed_waveforms` group in the given directory in the
             hdf file. If `None`, `compressed_waveforms` will be the root
             directory.
+        precision : {None, str}
+            Cast the saved parameters to the given precision before saving. If
+            None provided, will use whatever their current precision is. This
+            will raise an error if the parameters have single precision but the
+            requested precision is double.
         """
         if root is None:
             root = ''
         else:
             root = '%s/'%(root)
+        if precision is None:
+            precision = self.precision
+        elif precision == 'double' and self.precision == 'single':
+            raise ValueError("cannot cast single precision to double")
+        outdtype = _real_dtypes[precision]
         group = '%scompressed_waveforms/%s' %(root, str(template_hash))
```

Comment:
@cdcapano Line 630 sets `self.precision=precision` . Thus, when CompressedWaveform is generated by `compressed_waveform` it would save `self.precision` to be the same as the user input `args.precision` from `pycbc_compress_bank` . Again, here in `write_to_hdf`, `precision` would be the same as `args.precision` from `pycbc_compress_bank`. So, would it help to compare `precision` and `self.precision` , because aren't they going to be the same always? So instead, how about replacing lines 762-765 with just :
```
if precision is None:
    precision = 'single'
```
The compressed waveform points would be generated as `double` by default. So if the user provides `--precision` as `single` in `pycbc_compress_bank`, that gets passed as an argument to `write_to_hdf` and the compressed points get converted to `single` from `double` in line 766 above, when deciding `outdtype`. If the user provides `--precision` as `double`, points get saved as `double` and if `--precision` is `None`, again convert points generated as `double` to `single`.